selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java

```diff
@@ -860,50 +860,49 @@ public void scrollToItemByPath(String path) {
   }
 
   /**
-   * Expands specified path in project explorer Path given in format like like below:
+   * Expands specified {@code path} in project explorer. Path given in format like below:
    * TestProject/src/main/java/org.eclipse.che.examples
    *
    * @param path path for expand
    */
   public void expandPathInProjectExplorer(String path) {
-    String[] pathItems = path.split("/");
-    String currentItem = pathItems[0].replace('.', '/');
-    String currentPath = currentItem;
-
-    waitItem(currentPath);
-    openItemByPath(currentPath);
-    for (int i = 1; i < pathItems.length; i++) {
-      currentItem = pathItems[i].replace('.', '/');
-      currentPath += "/" + currentItem;
-      waitItem(currentPath);
-      loader.waitOnClosed();
-      openItemByPath(currentPath);
+    List<String> itemsPaths = getItemsPathsWithReplaceDots(path);
+    ListIterator<String> iterator = itemsPaths.listIterator();
+
+    while (iterator.hasNext()) {
+      String currentItem = iterator.next();
+      if (!iterator.hasNext()) {
+        waitItem(currentItem);
+        openItemByPath(currentItem);
+        break;
+      }
+
+      By nextItemLocator =
+          By.xpath(format(PROJECT_EXPORER_ITEM_TEMPLATE, itemsPaths.get(iterator.nextIndex())));
+      if (!seleniumWebDriverHelper.isVisible(nextItemLocator)) {
+        waitItem(currentItem);
+        openItemByPath(currentItem);
+      }
     }
   }
 
   /**
-   * Expands specified path in project explorer from defined item (will be try to open from defined
-   * item in path) Path given in format like like below:
-   * TestProject/src/main/java/org.eclipse.che.examples
+   * Parses {@code path} and makes full path for each element which separated by "/".
    *
-   * @param path path for expand
-   * @param numItem
+   * @param path Path in format "RootFolder/src/main/java/org.package".
+   * @return Paths in format ["RootFolder", "RootFolder/src", "RootFolder/src/main",
+   *     "RootFolder/src/main/java", "RootFolder/src/main/java/org/package"]
    */
-  public void expandPathInProjectExplorer(String path, int numItem) {
-    String[] pathItems = path.split("/");
-    String currentItem = "";
-    for (int i = 0; i < numItem + 1; i++) {
-      currentItem += "/" + pathItems[i].replace('.', '/');
-    }
-    String currentPath = currentItem.substring(1);
-    openItemByPath(currentPath);
-    for (int i = numItem + 1; i < pathItems.length; i++) {
-      currentItem = pathItems[i].replace('.', '/');
-      currentPath += "/" + currentItem;
-      waitItem(currentPath);
-      loader.waitOnClosed();
-      openItemByPath(currentPath);
+  private List<String> getItemsPathsWithReplaceDots(String path) {
```

Comment:
`getItemsPathsWithReplaceDots` doesn't clear explain the propose of method. 
What about another name: `splitPathOnSubitems`?

Reply:
:+1: